dart/lib/src/hub.dart

```diff
@@ -309,43 +314,95 @@ class Hub {
     }
   }
 
-  Map<String, String> traceHeaders() {
-    return {'sentry-trace': ''};
-  }
+  // Map<String, String> traceHeaders() {
+  //   return {'sentry-trace': ''};
+  // }
+
+  ISentrySpan startTransaction(
+    String name,
+    String operation, {
+    String? description,
+    bool? bindToScope,
+  }) =>
+      startTransactionWithContext(
+        SentryTransactionContext(
+          name,
+          operation,
+          description: description,
+        ),
+        bindToScope: bindToScope,
+      );
 
-  void startTransaction() {}
+  ISentrySpan startTransactionWithContext(
+    SentryTransactionContext transactionContext, {
+    Map<String, dynamic>? customSamplingContext,
+    bool? bindToScope,
+  }) {
+    if (!_isEnabled) {
+      _options.logger(
+        SentryLevel.warning,
+        "Instance is disabled and this 'startTransaction' call is a no-op.",
+      );
+    } else {
+      final item = _peek();
+
+      final samplingContext = SentrySamplingContext(
+          transactionContext, customSamplingContext ?? {});
+
+      transactionContext.sampled = _tracesSampler.sample(samplingContext);
+
+      final tracer = SentryTracer(transactionContext, this);
+
+      if (bindToScope ?? false) {
+        item.scope.span = tracer;
+      }
 
-  SentrySpan get span {
-    throw Exception();
+      return tracer;
+    }
+
+    return NoOpSentrySpan();
   }
 
+  // SentrySpan get span {
+  //   throw Exception();
+  // }
+
   Future<SentryId> captureTransaction(SentryTransaction transaction) async {
     var sentryId = SentryId.empty();
 
     if (!_isEnabled) {
       _options.logger(
         SentryLevel.warning,
-        "Instance is disabled and this 'captureEvent' call is a no-op.",
+        "Instance is disabled and this 'captureTransaction' call is a no-op.",
       );
     } else {
-      final item = _peek();
-      final event = await item.scope.applyToEvent(transaction.data, null);
-      if (event == null) {
-        return SentryId.empty();
+      if (!transaction.finished) {
+        _options.logger(
+          SentryLevel.warning,
+          'Capturing unfinished transaction: ${transaction.eventId}',
+        );
       }
```

Comment:
similar to https://github.com/getsentry/sentry-java/issues/1690 maybe, does it make sense to send a transaction if unfinished?

Reply:
thats the right behaviour, drop it